da-indexer/da-indexer-logic/src/celestia/indexer.rs

```diff
@@ -0,0 +1,206 @@
+use anyhow::{Error, Result};
+use celestia_rpc::prelude::*;
+use celestia_types::{blob::Blob, ExtendedHeader};
+use futures::{
+    stream,
+    stream::{repeat_with, BoxStream},
+    Stream, StreamExt,
+};
+use sea_orm::DatabaseConnection;
+use std::{collections::HashSet, sync::Arc, time::Duration};
+use tokio::{sync::RwLock, time::sleep};
+use tracing::instrument;
+
+use super::{
+    parser,
+    repository::{blobs, blocks},
+    rpc_client::Client,
+    settings::IndexerSettings,
+};
+
+#[derive(Debug)]
+struct Job {
+    height: u64,
+}
+
+pub struct Indexer {
+    client: Client,
+    db: Arc<DatabaseConnection>,
+    settings: IndexerSettings,
+
+    last_known_height: RwLock<u64>,
```

Comment:
It may be better and easier to use `AtomicU64` instead

Reply:
Yes, it looks better this way.